requests_kerberos/kerberos_.py

```diff
@@ -86,70 +84,75 @@ class HTTPKerberosAuth(AuthBase):
     object."""
     def __init__(
             self, mutual_authentication=REQUIRED,
-            service="HTTP", delegate=False):
+            service="HTTP", delegate=False, force_preemptive=False):
         self.context = {}
         self.mutual_authentication = mutual_authentication
         self.delegate = delegate
         self.pos = None
         self.service = service
+        self.force_preemptive = force_preemptive
 
-    def generate_request_header(self, response):
+    def generate_request_header(self, response, host, is_preemptive=False):
```

Comment:
Hypothetically, yes... 100% of my Kerberos experience has been in Windows bastard-kerberos-land, though, so I have no idea how kerberos functions in the rest of the world. ;)